### PR TITLE
React: modify task list

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 1]
```diff
@@ -19,53 +27,90 @@ const listsData = ['LIST', 'KANBAN BOARD', 'GANTT'];
...
-  const [itemVisibility, setItemVisibility] = useState(true);
-  const [data, setData] = useState([]);
+  const [gridData, setGridData] = useState([]);
+  const [filteredData, setFilteredData] = useState([]);
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     getTasks()
-      .then((data) => setData(data))
+      .then((data) => setGridData(data))
```
Comment (MikeVitik) [inline packages/react/src/pages/planning-task-list/planning-task-list.tsx:43]: Try to use Promise.all() and one catch

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 2]
```diff
@@ -19,53 +27,90 @@ const listsData = ['LIST', 'KANBAN BOARD', 'GANTT'];
...
+      navigate('/planning-task-details');
+    }
+  }, [list]);
 
   const refresh = useCallback(() => {
-    gridRef.current?.instance.refresh();
-  }, []);
+    if(list === listsData[0]) {
+      gridRef.current?.instance.refresh();
+    } else if(list === listsData[1]) {
```
Comment (MikeVitik) [inline packages/react/src/pages/planning-task-list/planning-task-list.tsx:72]: create const for list === listsData[1]

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 3]
```diff
@@ -19,53 +27,90 @@ const listsData = ['LIST', 'KANBAN BOARD', 'GANTT'];
...
 
   const showColumnChooser = useCallback(() => {
     gridRef.current?.instance.showColumnChooser();
   }, []);
 
   const exportToPDF = useCallback(() => {
-    const doc = new jsPDF();
-    exportDataGrid({
-      jsPDFDocument: doc,
+    if(list === listsData[0]) {
```
Comment (MikeVitik) [inline packages/react/src/pages/planning-task-list/planning-task-list.tsx:84]: introduce const list === listsData[0]

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 4]
```diff
@@ -19,53 +27,90 @@ const listsData = ['LIST', 'KANBAN BOARD', 'GANTT'];
 
 export const PlanningTaskList = () => {
   const gridRef = useRef<DataGrid>(null);
+  const kanbanRef = useRef<Sortable>(null);
+  const ganttRef = useRef<Gantt>(null);
+
+  const navigate = useNavigate();
 
   const [list, setList] = useState(listsData[0]);
```
Comment (MikeVitik) [inline packages/react/src/pages/planning-task-list/planning-task-list.tsx:35]: Replace for increace code readability
const [listItem, kanbanItem, ganttItem] = listData;
const [list, setList] = useState(listItem);

**Review by dxvladislavvolkov — DISMISSED** [review]

**Review by MikeVitik — DISMISSED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Review by MikeVitik — APPROVED** [review]

**Pull request merged**